app/src/cpp/CMakeLists.txt

```diff
@@ -2,7 +2,8 @@ cmake_minimum_required(VERSION 3.4.1)
 
 project ( cfc )
 
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -mllvm -inline-threshold=5000")
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -mllvm -inline-threshold=1500 -flto=thin")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -mllvm -inline-threshold=1500 -flto=thin")
```

Comment:
There's a lot that could be said about `-inline-threshold` (and inlining in general --  https://danlark.org/2020/10/08/i-need-extra-c-c-performance-now-how/ is a good read)

The gist for now is: 1500 is *hopefully* all we need, and is within the vague consensus of sane values for this compilation flag. 5000 was kind of crazy, "just inline everything you can", which worked on old build version but was probably not the best idea.

The motivation here is mostly the decoder code (Decoder/CimbReader/CimbDecoder), which benefits substantially from aggressive inlining.